pkg/telemetry/cassandra_agent/cassandra_agent_config_test.go

```diff
@@ -78,6 +166,47 @@ func Test_GetTelemetryAgentConfigMap(t *testing.T) {
 	assert.Equal(t, expectedCm.Namespace, cm.Namespace)
 }
 
+// Make sure we get default relabels when only endpoint is defined in spec.
+func Test_GetTelemetryAgentConfigMapWithDefinedEndpoint(t *testing.T) {
+	expectedCm := getExpectedConfigMap()
+	expectedCm.Data[filepath.Base(agentConfigLocation)] = endpointDefinedYaml
+	Cfg.RemoteClient = testutils.NewFakeClientWRestMapper() // Reset the Client
+	Cfg.TelemetrySpec = getExampleTelemetrySpec()
+	Cfg.TelemetrySpec.Cassandra.Relabels = nil
+	Cfg.TelemetrySpec.Cassandra.Endpoint = &telemetryapi.Endpoint{
+		Address: "192.168.1.10",
+		Port:    "50000",
+	}
+	cm, err := Cfg.GetTelemetryAgentConfigMap()
+	println(cm.Data)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedCm.Data["metric-collector.yaml"], cm.Data["metric-collector.yaml"])
+	assert.Equal(t, expectedCm.Name, cm.Name)
+	assert.Equal(t, expectedCm.Namespace, cm.Namespace)
+}
+
+func Test_GetTelemetryAgentConfigMapWithDefinedRelabels(t *testing.T) {
+	expectedCm := getExpectedConfigMap()
+	expectedCm.Data[filepath.Base(agentConfigLocation)] = relabelsDefinedYaml
+	Cfg.RemoteClient = testutils.NewFakeClientWRestMapper() // Reset the Client
+	Cfg.TelemetrySpec = getExampleTelemetrySpec()
+	Cfg.TelemetrySpec.Cassandra.Relabels = []promapi.RelabelConfig{
+		{
+			SourceLabels: []string{"tag1", "tag2"},
+			Separator:    ";",
+			Regex:        "(.*);(b.*)",
+			Action:       "drop",
+		},
+	}
+	Cfg.TelemetrySpec.Cassandra.Endpoint = nil
+	cm, err := Cfg.GetTelemetryAgentConfigMap()
+	println(cm.Data)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedCm.Data["metric-collector.yaml"], cm.Data["metric-collector.yaml"])
```

Comment:
The correct filename is ``/configs/metrics-collector.yaml``, notice the *s*

Reply:
That isn't what is specified [here](https://github.com/datastax/metric-collector-for-apache-cassandra/blob/master/config/metric-collector.yaml)?

Reply:
That's MCAC?

Reply:
Oops... I'm glad I mentioned this since I need to change some documentation back as I think it refers to MCAC, not the new collector. 

Thanks for clearing this up for me anyway, didn't realise this had also changed from the original PR.